actions/admin/upgrades/upgrade_comments_access.php

```diff
@@ -0,0 +1,74 @@
+<?php
+
+/**
+ * Convert comment annotations to entities
+ * 
+ * Run for 2 seconds per request as set by $batch_run_time_in_secs. This includes
+ * the engine loading time.
+ */
+// from engine/start.php
+global $START_MICROTIME;
+$batch_run_time_in_secs = 2;
+
+// if upgrade has run correctly, mark it done
+if (get_input('upgrade_completed')) {
+	// set the upgrade as completed
+	$factory = new ElggUpgrade();
+	$upgrade = $factory->getUpgradeFromPath('admin/upgrades/commentaccess');
+	if ($upgrade instanceof ElggUpgrade) {
+		$upgrade->setCompleted();
+	}
+
+	return true;
+}
+
+// Offset is the total amount of errors so far. We skip these
+// comments to prevent them from possibly repeating the same error.
+$offset = get_input('offset', 0);
+$limit = 50;
+
+$access_status = access_get_show_hidden_status();
+access_show_hidden_entities(true);
+
+$success_count = 0;
+$error_count = 0;
+
+do {
+	$dbprefix = elgg_get_config('dbprefix');
+	$options = array(
+		'type' => 'object',
+		'subtype' => 'comment',
+		'joins' => array(
+			"JOIN {$dbprefix}entities e2 ON e.container_guid = e2.guid"
+		),
+		'wheres' => array(
+			"e.access_id != e2.access_id"
+		),
+		'offset' => $offset,
+		'limit' => $limit
+	);
+			
+	$comments = elgg_get_entities($options);
+	
+	foreach ($comments as $comment) {
+		$container = $comment->getContainerEntity();
```

Comment:
ok, will do.  This is run with hidden status and access overridden, so there shouldn't be any case where the container can't be loaded unless there's a data problem where the container is missing or broken.  In that case what do you think we should do with it?